integration_test/ops_agent_test.go

```diff
@@ -456,31 +458,82 @@ func TestHTTPRequestLog(t *testing.T) {
 			t.Fatal(err)
 		}
 
+		// The HTTP request data that will be used in each log
 		httpRequestBody := map[string]interface{}{
 			"requestMethod": "GET",
 			"requestUrl":    "https://cool.site.net",
 			"status":        200,
 		}
-		logBody := map[string]interface{}{
-			"log":                                "4837291",
-			"logging.googleapis.com/httpRequest": httpRequestBody,
-		}
-		logBytes, err := json.Marshal(logBody)
-		if err != nil {
-			t.Fatalf("could not marshal test log: %v", err)
-		}
 
-		if err := gce.UploadContent(ctx, logger, vm, strings.NewReader(string(logBytes)+"\n"), logPath); err != nil {
-			t.Fatalf("error writing log line: %v", err)
-		}
+		// Write an http request log with to cloud logging and immediately query it
+		writeLog := func(httpRequestKey string, logId string) *cloudlogging.Entry {
+			logBody := map[string]interface{}{
+				"logId":        logId,
+				httpRequestKey: httpRequestBody,
+			}
+			logBytes, err := json.Marshal(logBody)
+			if err != nil {
+				t.Fatalf("could not marshal test log: %v", err)
+			}
+			err = gce.UploadContent(
+				ctx,
+				logger,
+				vm,
+				strings.NewReader(string(logBytes)+"\n"),
+				logPath)
+			if err != nil {
+				t.Fatalf("error writing log line: %v", err)
+			}
 
-		entry, err := gce.QueryLog(ctx, logger.ToMainLog(), vm, "mylog_source", time.Hour, "", gce.QueryMaxAttempts)
-		if err != nil {
-			t.Fatal(err)
-		}
-		if entry.HTTPRequest == nil {
-			t.Fatal("expected log entry to have HTTPRequest field")
-		}
+			entry, err := gce.QueryLog(
+				ctx,
+				logger.ToMainLog(),
+				vm,
+				"mylog_source",
+				time.Hour,
+				"jsonPayload.logId="+logId,
```

Comment:
Interesting, I didn't know it needed to be quoted. Seemed to work fine without it. Changed in [c40c623](https://github.com/GoogleCloudPlatform/ops-agent/pull/312/commits/c40c6235586cace5423a56bb7dc8aa3aabbfbc0c)

Reply:
The values you currently use in the calls to this function don't have to be quoted, but this is future-proofing the code, in case you decide on more complex log ids later.